kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/Metadata.java

```diff
@@ -164,6 +180,21 @@ public Map<String, String> getConfiguration() {
         return Collections.unmodifiableMap(configuration.get());
     }
 
+    /**
+     * Filter out the key-value pair matches exactly with the old properties.
+     *
+     * @param newProperties the new properties to be filtered
+     *
+     * @return the filtered properties
+     */
+    public Map<String, String> filterOutUnchangedProperties(Map<String, String> newProperties) {
+        Map<String, String> oldProperties = getConfiguration();
+        return newProperties.entrySet().stream()
+                .filter(entry -> !oldProperties.containsKey(entry.getKey()) ||
```

Comment:
TableConfig validation is case insensitive so shouldn't this also be case insensitive?

Reply:
The input of this function is the validated properties returned by the validation function whose key is same as the one in `TableConfig`. And the property name in Metadata is also stored in the case from `TableConfig`